test/e2e/spec/oidc_provider/client_spec.js

```diff
@@ -0,0 +1,67 @@
+describe("OpenId Provider", function() {
+    
+    var chakram = require('/usr/local/lib/node_modules/chakram/lib/chakram.js');
+    var expect = chakram.expect;
+    var DOMAIN = 'https://' + process.env.BOILERPLATE_DOMAIN;
+    var URL = DOMAIN + '/api/openid';
+
+    describe("provides an openid-configuration", function() {
+
+        it("as json", function() {
+            var response = chakram.get(URL + '/.well-known/openid-configuration');
+            return expect(response).to.have.header('Content-Type', 'application/json');
+        }); 
+
+        it("with provider information", function() {
+            var response = chakram.get(URL + '/.well-known/openid-configuration');
+            expect(response).to.have.json('issuer', URL);
+            expect(response).to.have.json('authorization_endpoint', URL + '/authorize');
+            expect(response).to.have.json('jwks_uri', URL + '/jwks');
+            return chakram.wait();
+        }); 
+    });
+
+    describe("reacts on an authorization request", function () {
+
+        it("with redirect to login page if client_id, redirect_uri, state, once, response_type and scope exists and valid", function () {
+            let response_type = process.env.RESPONSE_TYPE;
+            let scope = process.env.SCOPE;
+            let client_id = process.env.CLIENT_ID;
+            let redirect_uri = process.env.REDIRECT_URL;
+            let state = '32os93klas8923hf9qpw98qhf9f2q2q093fu';
+            let nonce = '2389fhq3034ht84f809hj348t943w94fcpm4';
+            let request = URL + '/authorize'
+                + '?response_type=' + response_type
+                + '&scope=' + scope
+                + '&nonce=' + nonce
+                + '&client_id=' + client_id 
+                + '&redirect_uri=' + redirect_uri
+                + '&state=' + state;
+            return chakram.get(request)
+            .then(function(respObj) {
+                let path = respObj.response.req.path;
+                expect(path).to.match(/^\/api\/openid\/login\/\?next=\/api\/openid\/authorize/);
+            });
+        });
+        
+        it("with no redirect if attributes missing or invalid", function () {
+            let response_type = process.env.RESPONSE_TYPE;
+            let scope = process.env.SCOPE;
+            let redirect_uri = process.env.REDIRECT_URL;
+            let state = '32os93klas8923hf9qpw98qhf9f2q2q093fu';
+            let nonce = '2389fhq3034ht84f809hj348t943w94fcpm4';
+            var request = URL + '/authorize'
+                + '?response_type=' + response_type
+                + '&scope=' + scope
+                + '&nonce=' + nonce
+                + '&client_id=1'
+                + '&redirect_uri=' + redirect_uri
+                + '&state=' + state;
+            return chakram.get(request)
+            .then(function(respObj) {
```

Comment:
use `expect(chakram.get(...)).to.have.header('Location').to.not.match(...)` ([example usage](https://github.com/nils-wisiol/django-rest-boilerplate/blob/20170825_e2e/test/e2e/spec/www_spec.js#L35), second argument can be a regular expression; don't forget to set `followRedirect: false` (ask @larissazech for details))